plugins/fileservices/fileservices.cpp

```diff
@@ -170,6 +170,30 @@ static IConstWorkUnit * getWorkunit(ICodeContext * ctx)
     return factory->openWorkUnit(wuid);
 }
 
+static WUState queryWorkunitState(ICodeContext * ctx, SCMStringBuffer &stateEx)
+{
+    Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+    Owned<IConstWorkUnit> wu(factory->openWorkUnit(ctx->getWuid()));
+    WUState state = WUStateRunning;
+    if (wu)
+    {
+        state = wu->getState();
+        wu->getStateEx(stateEx);
+    }
+    return state;
+}
+
+static void setWorkunitState(ICodeContext * ctx, WUState state, const char * msg)
+{
+    Owned<IWorkUnit> wu = ctx->updateWorkUnit();
+    if (wu)
+    {
+        wu->setState(state);
+        wu->setStateEx(msg);
```

Comment:
No, I have already changed the stateEx to "Blocked by fileservice activity: %s" so it needs to be reset